src/common/util.cpp

```diff
@@ -195,6 +195,57 @@ namespace tools
     catch (...) {}
   }
 
+  file_locker::file_locker(const std::string& filename)
+  {
+#ifdef WIN32
+    m_fd = INVALID_HANDLE_VALUE;
+    WCHAR wide_path[1000];
+    if (MultiByteToWideChar(CP_UTF8, 0, filename.c_str(), filename.size() + 1, wide_path, 1000))
+    {
+      m_fd = CreateFileW(wide_path, GENERIC_READ, 0, NULL, OPEN_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+      if (m_fd != INVALID_HANDLE_VALUE)
+      {
+        OVERLAPPED ov;
+        memset(&ov, 0, sizeof(ov));
+        if (!LockFileEx(m_fd, LOCKFILE_FAIL_IMMEDIATELY | LOCKFILE_EXCLUSIVE_LOCK, 0, 1, 0, &ov))
+        {
+          CloseHandle(m_fd);
+          m_fd = INVALID_HANDLE_VALUE;
+        }
+      }
+    }
+#else
+    m_fd = open(filename.c_str(), O_RDONLY | O_CREAT, 0666);
+    if (m_fd != -1)
+    {
+      if (flock(m_fd, LOCK_EX | LOCK_NB) == -1)
+      {
+        close(m_fd);
```

Comment:
This could do with a MERROR/MWARNING so weird stuff afterwards can be known to be coming from that.

Reply:
ok